pkg/snapshot/common.go

```diff
@@ -0,0 +1,63 @@
+package snapshot
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// getVolumeID returns the cloud provider specific identifier for the PersistentVolume.
+func getVolumeID(unstructuredPV runtime.Unstructured) (string, error) {
+	pv := new(v1.PersistentVolume)
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstructuredPV.UnstructuredContent(), pv); err != nil {
+		return "", errors.WithStack(err)
+	}
+	if pv.Spec.CSI != nil {
+		driver := pv.Spec.CSI.Driver
+		if driver == pxdDriverName {
+			return pv.Spec.CSI.VolumeHandle, nil
+		}
+		logrus.Infof("Unable to handle CSI driver: %s", driver)
+	}
+
+	if pv.Spec.PortworxVolume != nil {
+		if pv.Spec.PortworxVolume.VolumeID == "" {
+			return "", errors.New("portworx volumeID not found")
+		}
+		return pv.Spec.AWSElasticBlockStore.VolumeID, nil
```

Comment:
handle nil check for `AWSElasticBlockStore`

Reply:
I removed this file in a subsequent commit and moved the common logic to plugin.go itself. The AWSElasticBlockStore reference is not present anymore.

Reply:
👍  I reviewed PR commit by commits ,since its too big to load